RecoLocalCalo/HcalRecAlgos/src/MahiFit.cc

```diff
@@ -155,26 +155,32 @@ void MahiFit::doFit(std::array<float,3> &correctedOutput, int nbx) const {
   if (dynamicPed_) nnlsWork_.bxs[nnlsWork_.nPulseTot-1] = pedestalBX_;
 
   nnlsWork_.pulseMat.setZero(nnlsWork_.tsSize,nnlsWork_.nPulseTot);  
+  nnlsWork_.pulseDerivMat.setZero(nnlsWork_.tsSize,nnlsWork_.nPulseTot);
+
+  std::array<FullSampleVector, MaxPVSize> pulseShapeArray;
```

Comment:
there is no need for the array anymore actually....
just remove [iBX] as well for these two...